components/script/dom/svgelement.rs

```diff
@@ -2,26 +2,83 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+use dom::attr::Attr;
+use dom::bindings::js::LayoutJS;
+use dom::bindings::cell::DOMRefCell;
+use dom::bindings::codegen::Bindings::WindowBinding::WindowBinding::WindowMethods;
 use dom::bindings::inheritance::Castable;
+use dom::bindings::reflector::DomObject;
+use dom::bindings::str::DOMString;
 use dom::document::Document;
-use dom::element::Element;
+use dom::element::{AttributeMutation, Element};
 use dom::virtualmethods::VirtualMethods;
 use dom_struct::dom_struct;
 use html5ever::{LocalName, Prefix};
+use std::cell::RefMut;
+use style_traits::PARSING_MODE_ALLOW_UNITLESS_LENGTH;
+use style::properties::declaration_block::{Importance, PropertyDeclarationBlock};
+use style::properties::{parse_one_declaration_into, SourcePropertyDeclaration};
 use style::element_state::ElementState;
+use style::properties::PropertyId;
 
 #[dom_struct]
 pub struct SVGElement {
     element: Element,
+    presentation_attributes: DOMRefCell<PropertyDeclarationBlock>
```

Comment:
You probably want to use `Arc<Locked<PropertyDeclarationBlock>>`, and use the document write guard to modify it in place. That way passing it to style is just a refcount bump.